tests/util/translator/translator.spec.tsx

```diff
@@ -0,0 +1,498 @@
+                        children: {
+                            ["mockElement"]: {
+                                label: "Email address",
+                                uiType: "email-field",
+                                columns: {
+                                    desktop: 12,
+                                    tablet: 8,
+                                    mobile: 4,
+                                },
+                                placeholder: "",
+                                validation: [
+                                    {
+                                        required: true,
+                                        errorMessage:
+                                            "Email address is required",
+                                    },
+                                ],
+                            },
+                        },
+                        uiType: "grid",
+                    },
+                    "submit-button": {
+                        disabled: "invalid-form",
+                        label: "Submit",
+                        uiType: "submit",
+                    },
+                },
+                uiType: "section",
+            },
+        },
+    },
+};
+
+const MOCK_TEXT_BASED_ELEMENT: TElementMap = {
+    mock456: {
+        internalId: "mock456",
+        type: EElementType.TEXT,
+        id: "mockElement",
+        required: true,
+        requiredErrorMsg: "Input is required",
+        columns: { desktop: 12, tablet: 8, mobile: 4 },
+        placeholder: "",
+        label: ELEMENT_BUTTON_LABELS[EElementType.TEXT],
+    },
+};
+
+const MOCK_TEXT_BASED_SCHEMA = {
+    prefill: {},
+    schema: {
+        sections: {
+            section: {
+                children: {
+                    grid: {
+                        children: {
+                            ["mockElement"]: {
+                                label: "Short text",
+                                uiType: "text-field",
+                                columns: {
+                                    desktop: 12,
+                                    tablet: 8,
+                                    mobile: 4,
+                                },
+                                placeholder: "",
+                                validation: [
+                                    {
+                                        required: true,
+                                        errorMessage: "Input is required",
+                                    },
+                                ],
+                            },
+                        },
+                        uiType: "grid",
+                    },
+                    "submit-button": {
+                        disabled: "invalid-form",
+                        label: "Submit",
+                        uiType: "submit",
+                    },
+                },
+                uiType: "section",
+            },
+        },
+    },
+};
```

Comment:
instead of manually creating a mock for each test case, can we have a function generate that instead?

it can come with standard properties and allow overriding in areas where needed.

also, avoid the square brackets for keys as that is for dynamic keys only